pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -1174,7 +1174,7 @@ func (a *ApplicationBackupController) backupVolumes(backup *stork_api.Applicatio
 			backup.Status.FailedVolCount++
 		}
 	}
-	if len(backup.Status.Volumes) == backup.Status.FailedVolCount {
+	if (len(backup.Status.Volumes) != 0) && (len(backup.Status.Volumes) == backup.Status.FailedVolCount) {
```

Comment:
Should we also return from this block, after updating the CR as Failed and final?

Reply:
yes, if failing we can return directly from here. Will raise a separate PR for that as well as the finishTimeStamp one